samza-core/src/main/java/org/apache/samza/task/StreamOperatorTask.java

```diff
@@ -100,8 +103,11 @@ public final void process(IncomingMessageEnvelope ime, MessageCollector collecto
   }
 
   @Override
-  public final void window(MessageCollector collector, TaskCoordinator coordinator) throws Exception {
-    // TODO: invoke timer based triggers
+  public final void window(MessageCollector collector, TaskCoordinator coordinator)  {
+    context.getSystemStreamPartitions().forEach(ssp -> {
+        OperatorImpl impl = this.operatorGraph.get(ssp.getSystemStream());
+        impl.onTick(collector, coordinator);
```

Comment:
The Clock interface is in place now, but it uses the SystemClock.instance. We can certainly define `SettableClock` that has a `setCurrentTime` method and returns the cached timestamp instead of querying a `getCurrentTimeMillis()`. 

Are you fine if optimizations such as these are done as a follow-up to this PR?